### PR TITLE
Feature/sc 6646/delete a topic

### PR DESCRIPTION
**Pull request by d3n153**
# Description
    - Added a new scenario into the topic.feature
    - Added some methods into the topic-steps.js

## Links to Tickets or other pull requests
Base links to copy
- https://github.com/hpi-schul-cloud/integration-tests/pull/216
- https://ticketsystem.schul-cloud.org/browse/SC-6646

## Changes
- topic.feature and topic-steps.js were extended

## Deployment
nn

## New Repos, NPM pakages or vendor scripts
nn

## Approval for review
- [ ] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### Link to Definiton of Done
More and detailed information on the *definition of done* can be found [on Confluence](https://docs.schul-cloud.org/pages/viewpage.action?pageId=92831762)

### CODE REVIEW

**Review by pauljanns — DISMISSED** [review]
no remarks

File: features/topic.feature [hunk 1]
```diff
@@ -35,5 +35,22 @@ Feature: Create, edit and delete a topic in the course on the HPI SchulCloud pag
...
+	@deleteTopic
+	Scenario Outline: User delete a topic
+		Given <userRole> logs in with email '<email>' and password '<password>'
+		And <userRole> accepts data protection
+		When <userRole> goes to courses page
+		And <userRole> chooses course with name '<courseName>'
+		And <userRole> adds a topic with name '<topicName>'
+		And <userRole> adds content title '<contentTitle>' and text '<description>' to the topic content
+		Then <userRole> should see that created topic with name '<topicName>' is shown on the topic list
+		And <userRole> adds a new Topic with name '<secondTopicName>'
```
Comment (kgorner) [inline features/topic.feature:50]: Should use "Then" in this line and "And" in the following - both are things to do - no checks any more

File: page-objects/pages/coursePages/CRSSAddEditTopicPage.js [hunk 2]
```diff
@@ -72,11 +72,16 @@ async function addEtherpad(name, description) {
...
-	const resultMsg = ', List of task titles: ' + listOfTopicTitles;
-	expect(listOfTopicTitles, msg + resultMsg).to.include(name);
+async function isTopicOnTopicList(topicname, expectedValue) {
+	const allTopics = await topicTitleList();
+    const isTopicOnList = allTopics.some((element) => element.includes(topicname));
+    const fillString = !expectedValue ? 'not' : '';
+	const msg = `Topic with name is ${fillString} visible on the list: \n`;
+	const resultMsg = 'Expected: ' + topicname + ', Actual: ' + allTopics;
+
+	expectedValue
```
Comment (kgorner) [inline page-objects/pages/coursePages/CRSSAddEditTopicPage.js:82]: I need an explanation for that. Sorry:
`expectedValue
		? await expect(isTopicOnList, msg + resultMsg).to.equal(true)
		: await expect(isTopicOnList, msg + resultMsg).to.equal(false);`

File: step_definitions/topic-steps.js [hunk 3]
```diff
@@ -19,24 +20,39 @@ When(/^.* clicks on the topic with name '([^']*)'$/, function (topicName) {
...
 	await addEditTopicPage.setTopic(changedTopicName);
 });
+
 When(/^.* finds title '([^']*)' and changes title on '([^']*)' and text '([^']*)' of the topic$/, async function (contentTitle, changeContentTitle, changeDescription) {
 	await addEditTopicPage.setContentByTitle(contentTitle, changeContentTitle);
 	await addEditTopicPage.setDescription(changeDescription);
 	await addEditTopicPage.clickCreateTopicButton();
 });
+
+When(/^.* clicks on the trashcan button in the line of the topic with name '([^']*)' to delete the topic$/, async function (topicName) {
```
Comment (kgorner) [inline step_definitions/topic-steps.js:38]: in my opinion it is too long. My suggestion is:
clicks trashcan in line of topic named '([^']*)'

File: step_definitions/topic-steps.js [hunk 4]
```diff
@@ -19,24 +20,39 @@ When(/^.* clicks on the topic with name '([^']*)'$/, function (topicName) {
...
+	await GeneralCoursePage.clickOnTopicDeleteTrashcanButton(topicName);
+});
+
+When(/^.* clicks on the Löschen button in the Löschen pop up$/, async function () {
```
Comment (kgorner) [inline step_definitions/topic-steps.js:42]: in my opinion it is too long. My suggestion is:
clicks on Löschen button in Löschen pop up
=> Another question we need to agree on within the whole team is: Do we want to mix languages within the steps?

File: step_definitions/topic-steps.js [hunk 5]
```diff
@@ -19,24 +20,39 @@ When(/^.* clicks on the topic with name '([^']*)'$/, function (topicName) {
...
+});
+
 Then(/^.* should see changed topic with name '([^']*)' and content title '([^']*)' and description '([^']*)' is visible on the topic page$/,async function (changedTopicName, changedContentTitle, changedDescription) {
 	await addEditTopicPage.isTopicTitleVisible(changedTopicName);
 	await addEditTopicPage.isContentTopicTitleVisible(changedContentTitle);
 	await addEditTopicPage.isTopicDescriptionVisible(changedDescription);
-
 });
+
 Then(/^.* should see that edited topic with name '([^']*)' is shown on the topic list$/,async function (changedTopicName) {
```
Comment (kgorner) [inline step_definitions/topic-steps.js:52]: in my opinion it is too long. My suggestion is:
should see edited topic named '([^']*)'  on topic list

File: step_definitions/topic-steps.js [hunk 6]
```diff
@@ -19,24 +20,39 @@ When(/^.* clicks on the topic with name '([^']*)'$/, function (topicName) {
...
-	await addEditTopicPage.isTopicOnTopicList(changedTopicName);
+	await addEditTopicPage.isTopicOnTopicList(changedTopicName, true);
+});
 
+Then(/^.* should see that deleted topic with name '([^']*)' is not shown on the topic list$/,async function (topicName) {
```
Comment (kgorner) [inline step_definitions/topic-steps.js:56]: in my opinion it is too long. My suggestion is:
should not see deleted topic named '([^']*)' on topic list

**Review by kgorner — CHANGES_REQUESTED** [review]

File: page-objects/pages/coursePages/CRSSAddEditTopicPage.js (hunk 2 above)
Comment (d3n153) [reply page-objects/pages/coursePages/CRSSAddEditTopicPage.js:82]: This way we can handle the check of a new creation and edit with true and the deletion with false.
I took this from the courses page, where we use it in a similar way.

File: step_definitions/topic-steps.js (hunk 4 above)
Comment (d3n153) [reply step_definitions/topic-steps.js:42]: @Team let me know if I shall change to only english.

File: step_definitions/topic-steps.js (hunk 3 above)
Comment (d3n153) [reply step_definitions/topic-steps.js:38]: changed

File: step_definitions/topic-steps.js (hunk 5 above)
Comment (d3n153) [reply step_definitions/topic-steps.js:52]: changed

File: step_definitions/topic-steps.js (hunk 6 above)
Comment (d3n153) [reply step_definitions/topic-steps.js:56]: changed

File: page-objects/pages/coursePages/CRSSAddEditTopicPage.js [hunk 7]
```diff
@@ -73,11 +73,17 @@ async function addEtherpad(name, description) {
...
-	const msg = 'Topic with name [' + name + '] is not visible on the list \n';
-	const resultMsg = ', List of task titles: ' + listOfTopicTitles;
-	expect(listOfTopicTitles, msg + resultMsg).to.include(name);
+async function isTopicOnTopicList(topicname, expectedValue) {
+	const allTopics = await topicTitleList();
+    const isTopicOnList = allTopics.some((element) => element.includes(topicname));
+    const fillString = !expectedValue ? 'not' : '';
+	const msg = `Topic with name is ${fillString} visible on the list: \n`;
+	const resultMsg = 'Expected: ' + topicname + ', Actual: ' + allTopics; //Actual = Actual listed topics
+	//ggf. msg + resultsMsg zusammen packen vorher und dann mit neuer Variable übergeben
```
Comment (kgorner) [inline page-objects/pages/coursePages/CRSSAddEditTopicPage.js:82]: Sollte nicht mit gemerged werden :-)

**Review by kgorner — CHANGES_REQUESTED** [review]

**Review by kgorner — APPROVED** [review]
:-)

**Pull request merged**